repo/db/messages.go

```diff
@@ -0,0 +1,122 @@
+package db
+
+import (
+	"database/sql"
+	"encoding/json"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/OpenBazaar/openbazaar-go/pb"
+	"github.com/OpenBazaar/openbazaar-go/repo"
+)
+
+// MessagesDB - represents the messages table
+type MessagesDB struct {
+	modelStore
+}
+
+// NewMessageStore - return new MessagesDB
+func NewMessageStore(db *sql.DB, lock *sync.Mutex) repo.MessageStore {
+	return &MessagesDB{modelStore{db, lock}}
+}
+
+// Put - insert record into the messages
+func (o *MessagesDB) Put(messageID, orderID string, mType pb.Message_MessageType, peerID string, msg repo.Message) error {
```

Comment:
What are your thoughts here, @amangale?

Reply:
As of now the messageID generation is based on the orderID and messageType. This may not hold true forever. Hence, it is better to explicitly specify the messageID with the generation logic of messageID independent of Put.